.github/workflows/docs.yml

```diff
@@ -0,0 +1,187 @@
+name: Docs
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - "docs/**"
+      - "solidity/**"
+      - "!solidity/dashboard/**"
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  docs-detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      path-filter: ${{ steps.filter.outputs.path-filter }}
+    steps:
+      - uses: actions/checkout@v2
+        if: github.event_name == 'pull_request' 
+      - uses: dorny/paths-filter@v2
+        if: github.event_name == 'pull_request' 
+        id: filter
+        with:
+          filters: |
+            path-filter:
+              - './((docs|(solidity/!dashboard))/**)'
+
+  docs-tex:
+    runs-on: ubuntu-latest
+    container: keepnetwork/texlive:3
+    needs: docs-detect-changes
+    if: |
+      github.event_name != 'pull_request'
+        || needs.docs-detect-changes.outputs.path-filter == 'true'
+    defaults:
+      run:
+        working-directory: ./docs
+    env:
+      DOCS_DIR: ${{ github.workspace }}/tmp/docs
+    steps:
+      - uses: actions/checkout@v2
+
+      # Step generates files based on Makefile config. Current config produces
+      # a number of .png files, which are then used to generate 
+      # relay-states.pdf. We want to publish only the resulting .pdf file.
+      - name: Generate TeX document files
+        run: |
+          make clean
+          make docs
+          mkdir -p $DOCS_DIR
+          mv relay-states.pdf $DOCS_DIR
+
+      # A push event is a master merge; deploy to primary bucket.
+      - if: github.event_name == 'push'
+        name: Upload generated files
+        uses: thesis/gcp-storage-bucket-action@v3.1.0
+        with:
+          service-key: ${{ secrets.DOCS_KEEP_NETWORK_UPLOADER_SERVICE_KEY_JSON }}
+          project: cfc-production
+          bucket-name: docs.keep.network
+          bucket-path: core
```

Comment:
Agree that this should be the root for master push. We _should_ implement a similar build for ECDSA (it doesn't exist yet), and we _should_ make that push to `/ecdsa` for master push.

Ideally, we would do a docs build that would include `ecdsa`, `tbtc`, and `core` in one build---but the split repos make that very hard to deal with. We've done the `/tbtc` approach for tBTC, and I think we can do the same for ECDSA. We should continue to consider keep-core our “root” project for all intents and purposes, however.

Reply:
Changed to root directory, see 858826317d4776d487008b896b378d6b995c5446.